middleware/auth.js

```diff
@@ -1,17 +1,24 @@
 const jwt = require('jsonwebtoken');
+const { decrypt } = require('../util')
 
 function authenticateToken(req, res, next) {
   const authHeader = req.headers['authorization'];
   const token = authHeader && authHeader.split(' ')[1];
   if (token == null) return res.sendStatus(401);
-  jwt.verify(token, process.env.TOKEN_SECRET, (err, user) => {
 
-    if (err) return res.sendStatus(403);
-
-    req.user = user;
+  try {
+    const payload = jwt.verify(token, process.env.TOKEN_SECRET);
+    const decryptedIP = decrypt(payload.ip);
+    if (decryptedIP !== req.ip) {
+      res.status(403)
+      res.send({err:"Unauthorized"});
+    }
 
```

Comment:
add user to details to request when IP matches 
req.user= payload

Reply:
Done.